### PR TITLE
Automate certificate creation for ha-server deployment script

### PR DESCRIPTION
**Pull request by stefanos1316**
Have updated the script in order to automate the certificate creation and update.
I have executed the script on the remote host (185.70.186.134) and tested the following:

`$ curl -X GET http://epidose.ellak.gr/filter`

<html>
<head>301 Moved Permanently</head>
<body bgcolor="white">
<center>301 Moved Permanently</center>
<center>nginx/1.14.2</center>
</body>
</html><br \>

`$ curl -X GET https://epidose.ellak.gr/filter <br \>
Hi Stefanos, this works!`

Let me know if this is fine or changes are required.

### CODE REVIEW

File: epidose/back_end/deploy.yml [hunk 1]
```diff
@@ -95,3 +95,16 @@
...
+    - name: obtain certificate
+      become: yes
+      become_user: root
+      shell: certbot --nginx --non-interactive -d {{ server_name }} --agree-tos --email stefanos1316@gmail.com --email dds@aueb.gr
+
+    - name: add cron entry to renew cert every 12 hours
+      become: yes
+      become_user: root
+      cron:
+        name: update cert every 12 hours
```
Comment (dspinellis) [inline epidose/back_end/deploy.yml:108]: Every 4 days is more than enough

File: epidose/back_end/deploy.yml [hunk 2]
```diff
@@ -44,44 +47,41 @@
...
-    - name: add ha-server configurations
+    - name: add nginx configurations
+      tags: nginx
       become: yes
       become_user: root
       blockinfile: |
-        dest=/etc/nginx/sites-available/ha-server.conf
+        dest=/etc/nginx/conf.d/{{ server_name }}.conf
         content='server {
                     listen 80;
```
Comment (dspinellis) [inline epidose/back_end/deploy.yml:67]: Shouldn't this be the HTTPS port? Also, if you don't listen on 80, certificate renewal becomes much easier.

**Review by dspinellis — CHANGES_REQUESTED** [review]
Two small issues.

File: epidose/back_end/deploy.yml (hunk 1 above)
Comment (stefanos1316) [reply epidose/back_end/deploy.yml:108]: Christos from Eellak told me to invoke a renewal request every 12 hours,
but I will change it to four days as you said.

File: epidose/back_end/deploy.yml (hunk 2 above)
Comment (stefanos1316) [reply epidose/back_end/deploy.yml:67]: The HTTPS port is added automatically by the Certbox.
Please see the corresponding configuration file after the execution of the Ansible script:

```
# BEGIN ANSIBLE MANAGED BLOCK
server {
listen 80;
server_name epidose.ellak.gr;
root /home/epidose/epidose/epidose/back_end;
access_log /var/log/nginx_access.log;
error_log /var/log/nginx_error.log;

if ($scheme != "https") {
return 301 https://epidose.ellak.gr$request_uri;
}

location / {
proxy_set_header X-Forward-For $proxy_add_x_forwarded_for;
proxy_set_header Host $http_host;
proxy_redirect off;
proxy_pass http://185.70.186.134:5010;
}

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/epidose.ellak.gr/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/epidose.ellak.gr/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
# END ANSIBLE MANAGED BLOCK
```

For port 80, if invoked, a user will get a 301 Moved permanently message.
I am not sure if the renewal becomes easy or not.
But, I will try to remove `listen  80;`, run the Ansible script again, and see what happens.

File: epidose/back_end/deploy.yml (hunk 1 above)
Comment (stefanos1316) [reply epidose/back_end/deploy.yml:108]: Have changed to run every four days and this is the `crontab -e` output that I get:
`* * */4 * * /usr/bin/certbot renew --quiet && service nginx reload`
Is this okay?

File: epidose/back_end/deploy.yml (hunk 2 above)
Comment (stefanos1316) [reply epidose/back_end/deploy.yml:67]: I have tested the deploy script with the requested changes and even if I do not add the `listen 80` port, the certbot adds it:

```
    listen 80; # managed by Certbot

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/epidose.ellak.gr/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/epidose.ellak.gr/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
```

Also, this [article](https://letsencrypt.org/docs/allow-port-80/) from let's encrypt states that it is not a problem.
But I can definitely remove the line after the configurations are set.
What do you think?

**Comment by stefanos1316** [comment]
I haven't sent you yet the new changes that you request.
Please let me know if you are fine with my answers and then I will submit a new commit.

**Comment by dspinellis** [comment]
Yes, fine.

**Pull request merged**